conductor-client/src/main/kotlin/com/openlattice/data/storage/DataDeletionService.kt

```diff
@@ -116,41 +112,51 @@ class DataDeletionService(
     /* Authorization checks */
 
     @Timed
-    override fun authCheckForEntitySetAndItsNeighbors(entitySetId: UUID, deleteType: DeleteType, principals: Set<Principal>, entityKeyIds: Set<UUID>?) {
-        val isAssociationEntitySet = entitySetManager.getEntitySet(entitySetId)!!.flags.contains(EntitySetFlag.ASSOCIATION)
+    override fun authCheckForEntitySetsAndNeighbors(entitySetIds: Set<UUID>, deleteType: DeleteType, principals: Set<Principal>, entityKeyIds: Set<UUID>?) {
+        val srcDstEntitySets = entitySetIds.filter {
+            !entitySetManager.getEntitySet(it)!!.flags.contains(EntitySetFlag.ASSOCIATION)
+        }.toSet()
+        val neighborEdgeEntitySets = if (srcDstEntitySets.isEmpty()) mutableSetOf() else graphService.getNeighborEdgeEntitySets(srcDstEntitySets, entityKeyIds)
```

Comment:
One difference between this and the old code is that `neighborEdgeEntitySets` has not gone through the permissions filter yet. See `getAuthorizedNeighborEntitySetsQuery`, though that query doesn't check for a specific permission, which seems bad.

Reply:
Doesn't authorizationManager.accessChecksForPrincipal accomplish this? The AclKey includes both the entity set id and property type ids

Reply:
In my opinion, no. It's a different check. It's checking property permissions, not entity set permissions. That being said, I'm not totally sure if it's necessary since it's strictly a delete on entities.

@geekbeast do you think it is necessary to check entity set permissions in addition to property permissions?